### ISSUE TITLE
Ajuda com Datatable 

### ISSUE THREAD

**Issue by vitocaio** [frontendbr/forum#186]
Ola senhores, eu possuo uma tabela criada utilizando a biblioteca datatable do jQuery, porém eu carrego um numero x de linhas enquanto na verdade existe um numero y, por exemplo, tenho um total de 100 linhas, mas só exibo 85, lembrando que faço o carregamento dos dados via ajax(método do datatable) , gostaria de saber como carrego os outros 15 itens, chamando novamente meu ajax.

segue o código utilizado:

dt = $("#tblabels").DataTable( {
      	"dom": "<'row'<'col-sm-6'f><'col-sm-6 no-padding'B>><'row'<'col-sm-12 no-padding'tr>><'row'<'col-sm-12'i>>",
        "processing": true,
        "serverSide": true,
      	"buttons": [{
            "className": "btn btn-sm btn-success",
            "text": "<i class=\"fa fa-fw fa-plus\"></i> New",
            "action": function(e, dt, node, config) {
              	var $inp     = $(".form-validate").find("input");
              	var $txtarea = $(".form-validate").find("textarea");
      			$inp.val("");
              	$txtarea.val("");
				$(".translate textarea").val("");

              	if($inp.is(":disabled")){
                	$inp.prop("disabled", false);
                }

              	$("#btn-remove-label").prop("disabled", true);
            }
        }],
        "ajax": "/api/studio/modules/" + moduleId + "/labels",
      	"scrollY": document.body.offsetHeight - 180,
      	"scroller": true,
    });

  O retorno disto é...
  Showing 73 to 81 of 81 entries (filtered from 137 total entries)

**Comment by LFeh** [comment]
Consegue ajudar @fdaciuk ?

**Comment by fdaciuk** [comment]
Não conheço o datatable, mas não ficaria mais fácil fazer o request, e depois jogar os resultados na tabela?

**Comment by vitocaio** [comment]
è um pouco mais complexo que isso, desenvolvo em um framework da empresa, possui uma funcionalidade que gera uma table automaticamente, onde eu posso editar o font do mesmo para customizar, mas ele só retorna uma quantidade limitada de dados, os demais dados, ou o resto das linhas da table, iria vir junto a um metodo que eu devo implementar que ira carregar o resto dos dados quando o usurário der um sroll na pagina, igual ao facebook entendi?

**Comment by fdaciuk** [comment]
Mas, via ajax, você tem acesso a todos esses dados? Esquecendo o datatable: se você fizer um request ajax, você consegue pegar todos os dados que precisa?

**Comment by vitocaio** [comment]
Então, na verdade é até um pouco estranho, o Ajax que uso não seria o tradicional, é um método do datatable que faz um request Ajax, mas que só retorna a quantidade de dados que cabe no height atual do datatable, por exemplo, quanto maior o heigth do datatable mais dados ele retorna, mas não posso mudar o heigth pois isso vai gerar uma rolagem desnecessária, eu preciso executar algum método do datatable que quando dou um scroll na tela ele faz uma nova request retornando o resto dos dados.

**Comment by fdaciuk** [comment]
Sim, isso eu entendi.. mas você não pode fazer um request fora do datatable? Se não puder, tem que ler a doc do datatable pra ver como ele se comporta com relação a isso.

Mas se puder, seria só fazer o que você já está fazendo, mas ao invés de usar o ajax do datatable, na resposta do request, você popula o datatable, sacou?

**Comment by vitocaio** [comment]
pow, pode crer, vou tentar isso.

**Comment by RicardoJunior** [comment]
Opa! eu já trabalhei com o Datatables! @vitocaio

**Comment by vitocaio** [comment]
Eai, ja precisou fazer algo parecido com o que estou fazendo?

**Comment by RicardoJunior** [comment]
No datatables, ele trabalha com essa forma de paginação que só pode ser definida na criação (se eu não me engano), que é com uma propriedade "pageLength" (https://datatables.net/reference/option/pageLength). 

A muito tempo atrás, tive que criar uma forma de scroll infinito e nos plugins do datatable não existia nada. Eu não tinha muito tempo, então a solução criada foi o seguinte. Eu defini o maior número possível, tratei tudo pelo retorno do meu backend. Defini no backend um retorno de um número limitado de linhas, quando elas acabavam (após rolar no scroll), eu disparava o ajax de novo com o número de página + 1 (lembrando que a quantidade de paginas é o total de registros divido pela quantidade de registros que você definiu como inicial), e retorno das próximas chamadas ajax, você vai ter que trazer junto os resultados anteriores. Ele não guarda nenhum histórico interno do que já foi disparado. 

Algum tempo depois eu li num e-mail de release deles que eles tinham um novo plugin pra isso, que seria o: http://legacy.datatables.net/release-datatables/examples/basic_init/scroll_y_infinite.html

Acho que isso já ajuda!

**Comment by alexandreximenes** [comment]
estou tentando pesquisar uma coluna que é preenchido com <select> alguém sabe como fazer isso?

**Comment by AmandaFonseca** [comment]
Eu não entendi muito bem o que vc precisa mas eu estou estudando  dataTable agora também.
Mas acho que vc pode pegar carregar todos os valores via Ajax, adicionar os arquivos da biblioteca dataTable, incorporar o select que você quer, acho que o select da biblioteca chamado example_length vai servir já que vc pode alterar o css dele para deixa-lo parecido com aquele select do face. Ai você inspeciona esse elemento e altera o valor dele ou altera o valor dele via jquery.

**Comment by alexandreximenes** [comment]
Então amanda, Tenho 2 colunas no meu datatable, uma é um campo texto normal
e outra é um select <select> e quando vou pesquisar ele só pesquisa pela 1
coluna e não pela segunda também. quero poder pesquisar pelas 2

​colunas














Em 16 de junho de 2017 17:43, AmandaFonseca <notifications@github.com>
escreveu:

**Comment by AmandaFonseca** [comment]
Dá uma olhada nesses links :https://webix.com/widget/datatable/ 
https://webix.com/demo/datatable/multi-select-filter/

**Comment by alexandreximenes** [comment]
muito legal esse framework, mas eu não posso trocar por ser da empresa.
tenho que modificar naquele mesmo, vc tem ideia de como fazer?










2017-06-19 10:13 GMT-03:00 AmandaFonseca <notifications@github.com>:

**Comment by Bulgaricus** [comment]
@AmandaFonseca but i'm christian

**Comment by AmandaFonseca** [comment]
Teria que estudar a documentação mas infelizmente estou cheia de trabalho.
Lamento amigo.
